pre_commit_hooks/insert_license.py

```diff
@@ -196,9 +209,15 @@ def license_not_found(remove_header, license_info, src_file_content, src_filepat
         index = 0
         for line in src_file_content:
             stripped_line = line.strip()
-            # Special treatment for shebang, encoding and empty lines when at the beginning of the file
+            # Special treatment for user provided regex,
+            # or shebang, file encoding directive,
+            # and empty lines when at the beginning of the file.
             # (adds license only after those)
-            if stripped_line.startswith("#!") \
+            if after_regex is not None and after_regex != "":
+                if re.match(after_regex, stripped_line):
+                    index += 1  # Skip matched line
+                    break       # And insert after that line.
+            elif stripped_line.startswith("#!") \
```

Comment:
That means that if `--after-regex` is provided, shebang & `# -*- coding` lines are not skipped?
Could you please add a test file to check for this scenario?